tls/s2n_cipher_suites.c

```diff
@@ -1155,24 +1155,138 @@ int s2n_set_cipher_as_client(struct s2n_connection *conn, uint8_t wire[S2N_TLS_C
     return 0;
 }
 
-static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
+static int s2n_wire_ciphers_contain(const uint8_t *match, struct s2n_stuffer *wire_stuffer, uint32_t count, uint32_t cipher_suite_len)
 {
     for (uint32_t i = 0; i < count; i++) {
-        const uint8_t *theirs = wire + (i * cipher_suite_len) + (cipher_suite_len - S2N_TLS_CIPHER_SUITE_LEN);
+        uint8_t theirs[S2N_SSLv2_CIPHER_SUITE_LEN] = {0};
+        POSIX_GUARD(s2n_stuffer_read_bytes(wire_stuffer, theirs, cipher_suite_len));
 
         if (!memcmp(match, theirs, S2N_TLS_CIPHER_SUITE_LEN)) {
+            POSIX_GUARD(s2n_stuffer_reread(wire_stuffer));
             return 1;
         }
     }
 
+    POSIX_GUARD(s2n_stuffer_reread(wire_stuffer));
```

Comment:
If you're going to reset the stuffer on every call, why pass a stuffer to this method at all? This method could construct the stuffer itself. You could either leave the signature untouched or pass in a blob.